consensus/p2p/buffered/buffered_topic_subscription.go

```diff
@@ -0,0 +1,44 @@
+package buffered
+
+import (
+	"context"
+	"errors"
+
+	"github.com/NethermindEth/juno/utils"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+type TopicSubscription struct {
+	log        utils.Logger
+	bufferSize int
+	callback   func(context.Context, *pubsub.Message)
+}
+
+func New(log utils.Logger, bufferSize int, callback func(context.Context, *pubsub.Message)) TopicSubscription {
+	return TopicSubscription{
+		log:        log,
+		bufferSize: bufferSize,
+		callback:   callback,
+	}
+}
+
+func (b TopicSubscription) Loop(ctx context.Context, topic *pubsub.Topic) {
+	sub, err := topic.Subscribe(pubsub.WithBufferSize(b.bufferSize))
+	if err != nil {
+		b.log.Errorw("unable to subscribe to topic with error: %w", err)
+		return
+	}
+	defer sub.Cancel()
+
+	for {
+		msg, err := sub.Next(ctx)
+		if errors.Is(err, context.Canceled) {
+			return
+		}
+		if err != nil {
+			b.log.Errorw("unable to receive message", "error", err)
+			continue
```

Comment:
Another nitpick is to put the error `is` check inside the `if err != nil` block of code